src/codes/june_challenge/day9/IsSubsequence.cpp

```diff
@@ -0,0 +1,18 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+class Solution {
+public:
+    bool isSubsequence(string s, string t) {
+        int i = 0, j = 0;
+        while (i < s.length() && j < t.length()) {
+            if (s[i] == t[j]) {
+                i++;
+                j++;
+            } else {
+                j++;
```

Comment:
No j is required if you directly iterate over the characters of s

Reply:
`
int i = 0;
for (char c : s) {
  if (i < t.size() && c == t[i]) {
    i++:
  }
}
return i == t.size()
`
Something like this

Reply:
your approach making the code more elegant, still you're using two characters i and c to iterate.